### PR TITLE
feat: added minutes and countdown to timer

### PR DESCRIPTION
**Pull request by nordemaen**
- added 25/50 and custom duration
- countdown in time
- added validation of positive integer in field

### CODE REVIEW

File: src/components/timer.mjs [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
   #isPaused = false;
-  #timerDisplay; 
+  #timerDisplay;
   #startButton;
   #stopButton;
   #pauseButton;
+  #durationInput;
+  #durationOptions;
+  #selectedDuration;
+  #remainingTime;
```
Comment (xZa-k) [inline src/components/timer.mjs:13]: How would other components access this?

File: src/components/timer.mjs [hunk 2]
```diff
@@ -25,41 +35,72 @@ export class Timer extends HTMLElement {
...
+    if (this.#durationOptions.value === 'custom') {
+      this.#selectedDuration = parseInt(this.#durationInput.value);
+      if (isNaN(this.#selectedDuration) || this.#selectedDuration < 1) {
+        alert('Please enter a valid duration (positive integer).');
+        return;
+      }
+    }
+
+    this.#remainingTime = this.#selectedDuration * 60;
     this.#intervalId = setInterval(() => {
```
Comment (xZa-k) [inline src/components/timer.mjs:72]: The callback runs for the first time after a second, so maybe seperate the callback so it starts instantly.
```suggestion
    let intervalCallback = () => {
      if (!this.#isPaused) {
        if (this.#remainingTime > 0) {
          this.#remainingTime--; 
          this.#updateDisplay();
        } else {
          this.stop(); 
        }
      }
    };
    intervalCallback();
    this.#intervalId = setInterval(intervalCallback);
```

**Review by xZa-k — CHANGES_REQUESTED** [review]
Could be a good idea to make some of the private fields public, and change the interval for the timer to run instantly, not after the first second (maybe even have the callback as a field)

**Review by xZa-k — APPROVED** [review]

File: src/components/timer.mjs (hunk 1 above)
Comment (nordemaen) [reply src/components/timer.mjs]: refer to #6

**Pull request merged**